crates/matrix-sdk/src/http_client.rs

```diff
@@ -121,87 +121,59 @@ impl HttpClient {
             None => self.request_config,
         };
 
-        let request = if !self.request_config.assert_identity {
-            self.try_into_http_request(request, session, config).await?
-        } else {
-            self.try_into_http_request_with_identity_assertion(request, session, config).await?
-        };
+        let auth_scheme = Request::METADATA.authentication;
+        if !matches!(auth_scheme, AuthScheme::AccessToken | AuthScheme::None) {
+            return Err(HttpError::NotClientRequest);
+        }
 
-        self.inner.send_request(request, config).await
-    }
+        let access_token;
 
-    async fn try_into_http_request<Request: OutgoingRequest>(
-        &self,
-        request: Request,
-        session: Arc<RwLock<Option<Session>>>,
-        config: RequestConfig,
-    ) -> Result<http::Request<Bytes>, HttpError> {
-        let read_guard;
-        let access_token = if config.force_auth {
-            read_guard = session.read().await;
-            if let Some(session) = read_guard.as_ref() {
-                SendAccessToken::Always(session.access_token.as_str())
-            } else {
+        let request = if !self.request_config.assert_identity {
+            let send_access_token = if matches!(auth_scheme, AuthScheme::None) && !config.force_auth
```

Comment:
Is there a point of using `matches!()` vs `==` if you're matching a single enum variant? A `==` would be shorter as well.